### PR TITLE
Euclidean Algorithm and Bézout Identity for Integers

### PR DESCRIPTION
**Pull request by kangrongji**
This PR contains what mentioned in the title. If anyone defines the concept of Bézout domain, the proof can be immediately modified to show `ℤ` is among them.

I'm on half way to prove the existence of Smith normal form, based on this little PR.

### CODE REVIEW

**Comment by mortberg** [comment]
Nice! For some reason I thought there was already some divisibility theory in the library, but that's maybe for naturals? I'll just take a quick look now and comment more once I have time for a proper readthrough

File: Cubical/Data/Int/Divisibility.agda [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+-- Bézout and Euclidean Domain
+
+record Bézout (m n : ℤ) : Type where
+  constructor bezout
+  field
+    coef₁ : ℤ
+    coef₂ : ℤ
+    gcd   : ℤ
+    identity : coef₁ · m + coef₂ · n ≡ gcd
+    isGCD : (gcd ∣ m) × (gcd ∣ n)
```
Comment (mortberg) [inline Cubical/Data/Int/Divisibility.agda:119]: Don't you need to say that `gcd` is the biggest such divisor as well?

Comment (kangrongji) [reply Cubical/Data/Int/Divisibility.agda:119]: That's equivalent to `identity` + `isGCD` and I think this form is simpler and useful. But It seems the name `isGCD` not exactly precise. Should I include the definition of GCD and add a proof of equivalence?

**Comment by kangrongji** [comment]
Yes. The library only has that for natural numbers and no Bézout identity. And also the results are a bit chaotic, so I somewhat packed them up for integers.

File: Cubical/Data/Int/Divisibility.agda (hunk 1 above)
Comment (kangrongji) [reply Cubical/Data/Int/Divisibility.agda:201]: I add an equivalence proof.

File: Cubical/Data/Int/Divisibility.agda [hunk 2]
```diff
@@ -0,0 +1,436 @@
...
+
+module _
+  (b : Bézout m n) where
+
+  private
+    g = b .gcd
+
+  gcdIsGCD : k ∣ m → k ∣ n → k ∣ g
+  gcdIsGCD {k = k} p q =
+    subst (k ∣_) (b .identity) (∣-+ (∣-right· {n = b .coef₁} p) (∣-right· {n = b .coef₂} q))
```
Comment (kangrongji) [inline Cubical/Data/Int/Divisibility.agda:230]: I added an equivalence proof.

File: Cubical/Data/Int/Divisibility.agda (hunk 1 above)
Comment (mortberg) [reply Cubical/Data/Int/Divisibility.agda:201]: Right, maybe calling it `isDivisor` would be better?

File: Cubical/Data/Int/Divisibility.agda (hunk 2 above)
Comment (mortberg) [reply Cubical/Data/Int/Divisibility.agda:230]: Great!

**Review by mortberg — APPROVED** [review]
Just one small renaming needed and then I would be happy to merge

**Comment by kangrongji** [comment]
I rename it to `isCD` for isCommonDivisor, which is used in the files of `ℕ`. @mortberg

**Pull request merged**